### PR TITLE
Add the Data Validation help topic

### PR DESCRIPTION
**Pull request by RomanTsukanov**

### CODE REVIEW

File: docs/data-validation.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    "name": "memo",
+    "type": "comment",
+    // ...
+  }]
+};
+const survey = new Model(surveyJson);
+
+survey.onValidateQuestion.add((survey, options) => {
+  if (options.name === "memo") {
+    if (!!options.value.indexOf("survey")) {
```
Comment (tsv2013) [inline docs/data-validation.md:93]: May be it should be following condition
if (options.value.indexOf("survey") == -1) {

File: docs/data-validation.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+| `"regex"` | [`RegexValidator`](https://surveyjs.io/Documentation/Library?id=RegexValidator) | Throws an error if an entered value does not match a regular expression defined in the [`regex`](https://surveyjs.io/Documentation/Library?id=RegexValidator#regex) property. |
+
+- [View the "Built-In Validators" example](https://surveyjs.io/Examples/Library?id=validators-standard)
+- [View the "Expression Validator" example](https://surveyjs.io/Examples/Library?id=validators-expression) 
+
+## Implement Custom Client-Side Validation
+
+SurveyJS Library raises the [`onValidateQuestion`](https://surveyjs.io/Documentation/Library?id=surveymodel#onValidateQuestion) event that you can handle to add custom validation logic to your survey. For example, the following code checks that the answer to a `"memo"` question contains the word "survey":
```
Comment (tsv2013) [inline docs/data-validation.md:78]: Subsection? ### Validate on Event

File: docs/data-validation.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      complete();
+    });
+}
+
+survey.onServerValidateQuestions.add(validateCountry);
+```
+
+[View example](https://surveyjs.io/Examples/Library?id=validators-server)
+
+Alternatively, you can implement custom validation via [expressions](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#expressions). Create an [asynchronous function](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#asynchronous-functions), register it, and then call it from within your expression. The following code uses this approach to demonstrate the use case above:
```
Comment (tsv2013) [inline docs/data-validation.md:176]: Subsection? ### Validation using `expression` built-in validator

Comment (tsv2013) [reply docs/data-validation.md:176]: Or ### Using custom functions in validation

**Review by tsv2013 — CHANGES_REQUESTED** [review]

**Review by tsv2013 — APPROVED** [review]

File: docs/data-validation.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+survey.onValidateQuestion.add((survey, options) => {
+  if (options.name === "memo") {
+    if (options.value.indexOf("survey") === -1) {
+      options.error = 'Your answer must contain the word "survey"'
+    }
+  }
+});
+```
+
+Alternatively, you can implement custom validation via [expressions](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#expressions). Create a [custom function](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#custom-functions), register it, and then call it from within your expression. The following code uses this approach to demonstrate the use case above:
```
Comment (dmitry-eliseev-devexpress) [inline docs/data-validation.md:100]: http://tw.devexpress.devx/language-guidelines/misused-words/via/

```suggestion
Alternatively, you can use [expressions](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#expressions) to implement custom validation. Create a [custom function](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#custom-functions), register it, and then call it from within your expression. The following code uses this technique to implement the same value validation scenario:
```

File: docs/data-validation.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+      complete();
+    });
+}
+
+survey.onServerValidateQuestions.add(validateCountry);
+```
+
+[View example](https://surveyjs.io/Examples/Library?id=validators-server)
+
+Alternatively, you can implement custom validation via [expressions](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#expressions). Create an [asynchronous function](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#asynchronous-functions), register it, and then call it from within your expression. The following code uses this approach to demonstrate the use case above:
```
Comment (dmitry-eliseev-devexpress) [inline docs/data-validation.md:176]: http://tw.devexpress.devx/language-guidelines/misused-words/via/

```suggestion
Alternatively, you can use [expressions](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#expressions) to implement custom validation. Create an [asynchronous function](https://surveyjs.io/Documentation/Library?id=design-survey-conditional-logic#asynchronous-functions), register it, and then call it within your expression. The following code uses this technique to implement the previously demonstrated validation scenario:
```

**Review by dmitry-eliseev-devexpress — APPROVED** [review]

**Review by tsv2013 — APPROVED** [review]

File: docs/data-validation.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+| `"regex"` | [`RegexValidator`](https://surveyjs.io/Documentation/Library?id=RegexValidator) | Throws an error if an entered value does not match a regular expression defined in the [`regex`](https://surveyjs.io/Documentation/Library?id=RegexValidator#regex) property. |
+
+- [View the "Built-In Validators" example](https://surveyjs.io/Examples/Library?id=validators-standard)
+- [View the "Expression Validator" example](https://surveyjs.io/Examples/Library?id=validators-expression) 
+
+## Implement Custom Client-Side Validation
+
+SurveyJS Library raises the [`onValidateQuestion`](https://surveyjs.io/Documentation/Library?id=surveymodel#onValidateQuestion) event that you can handle to add custom validation logic to your survey. For example, the following code checks that the answer to a `"memo"` question contains the word "survey":
```
Comment (annnke) [inline docs/data-validation.md:78]:
```suggestion
SurveyJS Library raises the [`onValidateQuestion`](https://surveyjs.io/Documentation/Library?id=surveymodel#onValidateQuestion) event, which you can handle to add custom validation logic to your survey. For example, the following code checks that the answer to a `"memo"` question contains the word "survey":
```

File: docs/design-survey-conditional-logic.md [hunk 7]
```diff
@@ -426,12 +426,16 @@ function myFunc(params) {
 After you implement a custom function, register it in the `FunctionFactory`:
```
Comment (annnke) [inline docs/design-survey-conditional-logic.md:426]:
```suggestion
After you implement a custom function, register it in `FunctionFactory`:
```

**Review by annnke — APPROVED** [review]

**Pull request merged**